Automattic/jetpack
# Assistant: Add VaultPress card to the assistant flow

Context: 1201210512993648-as-1203060617058104

## Proposed changes:

This PR changes the Assistant flow for free users. The following changes were added:

* Added two different cards to the Assistant flow: one highlighting VaultPress for free users, the other one more focused on WooCommerce stores
* Updated the e2e tests to cover the changes on the Assistant
* Updated how the list style in `descriptionList` is displayed
* Conditionally render [the description paragraph](https://github.com/Automattic/jetpack/blob/c5d6a4b3b83c8b8ce71a736e852cbd58d172e56d/projects/plugins/jetpack/_inc/client/recommendations/prompts/prompt-layout/index.jsx#L79) to avoid the element and its style if a description is not available
* Added a quick helper function to calculate the monthly price of the product. That will change in the future (see pbNhbs-53E-p2)

Note: the CTA text on the card page (i.e. "Try for $1 for 30 days") is dynamic. It's formatted to your currency settings, and might change if there's an intro offer or not.

### Other information:


## Jetpack product discussion



pbNhbs-4X5-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






### Testing the flow for free users

* Spin up an ephemeral/JN site with Jetpack Beta included (so you can test this branch)
* Connect your site
* Choose a free plan on the pricing page
* Continue to the Assistant; make sure the last screen before the Summary is the VaultPress card
* Confirm that the card looks like the screenshots below, and it matches the design (some design changes were not applied to not mess other steps in the Assistant flow)
* Make sure the CTA text shows the right and localized price
  * If you click on the CTA, it will redirect you to the checkout page. After purchase, it will redirect back to the feature onboarding flow
  * If you click "Not now", you should be redirected to recommendations/summary

### Testing the flow for free users who have WooCommerce installed and active
* Follow the test plan described above, but in the first step, spin up a site with both Jetpack Beta **AND** WooCommerce. Make sure to activate WooCommerce before proceeding
* Follow the test above and confirm that the design and copy look like what's in the screenshot below

### Testing the flow for paid users
* The VaultPress Backup card should not show at all. You can test it going through the steps above, but purchasing a product right after connecting Jetpack (on the pricing page)

### Screenshots
|  Without WooCommerce |  With WooCommerce |
|-------------|----------|
| <img width="1046" alt="image" src="https://user-images.githubusercontent.com/5714212/216740238-1a13c378-00a9-4b74-9ffa-915febece9bc.png"> | <img width="1049" alt="image" src="https://user-images.githubusercontent.com/5714212/216740169-d77ef0cf-e3f2-45b0-ad65-2ffb22c5786e.png"> |
